doc/ntp/ntp-design.md

```diff
@@ -0,0 +1,898 @@
+<!-- omit in toc -->
+# SONiC Network Time Protocol (NTP) client configuration
+
+## High Level Design document
+
+## Table of contents
+- [Revision](#revision)
+- [About this manual](#about-this-manual)
+- [Scope](#scope)
+- [1 Introduction](#1-introduction)
+  - [1.1 Feature overview](#11-feature-overview)
+  - [1.2 Requirements](#12-requirements)
+- [2 Design](#2-design)
+  - [2.1 Overview](#21-overview)
+  - [2.2 NTP daemon configuration](#22-ntp-daemon-configuration)
+  - [2.3 Configuration agent](#23-configuration-agent)
+  - [2.4 DB schema](#24-db-schema)
+  - [2.5 Flows](#25-flows)
+  - [2.6 CLI](#26-cli)
+      - [2.6.2.1 Config command group](#2621-config-command-group)
+      - [2.6.2.2 Show command group](#2622-show-command-group)
+  - [2.7 YANG model](#27-yang-model)
+  - [2.8 Warm/Fast boot](#28-warmfast-boot)
+- [3 Test plan](#3-test-plan)
+  - [3.1 Unit tests via VS](#31-unit-tests-via-vs)
+
+# Revision
+| Rev | Date       | Author            | Description     |
+|:---:|:----------:|:-----------------:|:----------------|
+| 0.1 | 01/02/2023 | Yevhen Fastiuk 🇺🇦  | Initial version |
+
+# About this manual
+This document provides general information about NTP implementation in SONiC and NTP CLI.
+It describes high-level behavior, internal definition, design of the commands, syntax, and output definition.
+
+# Scope
```

Comment:
Please mention what is already available in the code and what's the scope of this HLD.